### ISSUE TITLE
Doesn't work even when inside a git repository

### ISSUE THREAD

**Issue by mr-rodgers** [pypa/setuptools-scm#89]
I'm not sure what's causing this; maybe you guys'll know better;

```
(env35) C:\Users\scrut\code\phi\auth [develop +0 ~4 -0 | +0 ~16 -0 !]> dir


    Directory: C:\Users\scrut\code\phi\auth


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        6/12/2016  12:38 PM                .git
d-----        6/12/2016  12:24 PM                .tox
d-----        6/11/2016   9:02 PM                doc
d-----        6/11/2016   9:41 PM                src
d-----        6/11/2016   9:02 PM                test
-a----        6/11/2016   9:02 PM            395 .gitattributes
-a----        6/12/2016  12:22 PM           1584 .gitignore
-a----        6/11/2016   9:02 PM            116 .travis.yml
-a----        6/11/2016   9:02 PM           1077 LICENSE.rst
-a----        6/11/2016   9:02 PM             40 MANIFEST.in
-a----        6/12/2016  12:09 PM            897 README.rst
-a----        6/12/2016  12:38 PM           1452 setup.py
-a----        6/11/2016   9:02 PM            610 tox.ini


(env35) C:\Users\scrut\code\phi\auth [develop +0 ~4 -0 | +0 ~16 -0 !]> python setup.py egg_info
Traceback (most recent call last):
  File "setup.py", line 58, in <module>
    entry_points={},
  File "c:\users\scrut\appdata\local\programs\python\python35-32\Lib\distutils\core.py", line 108, in setup
    _setup_distribution = dist = klass(attrs)
  File "C:\Users\scrut\code\env35\lib\site-packages\setuptools\dist.py", line 272, in __init__
    _Distribution.__init__(self,attrs)
  File "c:\users\scrut\appdata\local\programs\python\python35-32\Lib\distutils\dist.py", line 281, in __init__
    self.finalize_options()
  File "C:\Users\scrut\code\env35\lib\site-packages\setuptools\dist.py", line 327, in finalize_options
    ep.load()(self, ep.name, value)
  File "c:\users\scrut\code\phi\auth\.eggs\setuptools_scm-1.11.0-py3.5.egg\setuptools_scm\integration.py", line 19, in version_keyword
  File "c:\users\scrut\code\phi\auth\.eggs\setuptools_scm-1.11.0-py3.5.egg\setuptools_scm\__init__.py", line 104, in get_version
  File "c:\users\scrut\code\phi\auth\.eggs\setuptools_scm-1.11.0-py3.5.egg\setuptools_scm\__init__.py", line 82, in _do_parse
LookupError: setuptools-scm was unable to detect version for 'C:\\Users\\scrut\\code\\phi\\auth'.

Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.

For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
```

I only recently started seeing this after cloning the repo to a fresh setup.

**Comment by RonnyPfannschmidt** [comment]
looks like a windows platform issue

whats the output of `git describe` ?

**Comment by mr-rodgers** [comment]
Wait, what?

```
(env35) C:\Users\scrut\code\phi\auth [develop +0 ~4 -0 | +0 ~16 -0 !]> git describe
fatal: No names found, cannot describe anything.
```

...so very much not a setuptools_scm issue then (I'm pretty sure this repo has a few tags that I've previously set).

**Comment by RonnyPfannschmidt** [comment]
@te-je 
 please also take a look at the debug output and tell me the version of setuptools_scm you use, it should indicate a bit better what is wrong since you do have a git repo there

debug output can be obtained via SETUPTOOLS_SCM_DEBUG=1 in the environment (i have no idea how to set that on a windows terminal) in case of doubt

```
import os; os.environ['SETUPTOOLS_SCM_DEBUG'] ='1'
```

as first line in setup.py

**Comment by mr-rodgers** [comment]
Here it is:

```
root 'C:\\Users\\scrut\\code\\phi\\auth'
looking for ep setuptools_scm.parse_scm C:\Users\scrut\code\phi\auth
found ep .git = setuptools_scm.git:parse
cmd 'git rev-parse --show-toplevel'
out b'C:/Users/scrut/Code/phi/auth\n'
real root C:/Users/scrut/Code/phi/auth
Traceback (most recent call last):
  File "setup.py", line 58, in <module>
    entry_points={},
  File "c:\users\scrut\appdata\local\programs\python\python35-32\Lib\distutils\core.py", line 108, in setup
    _setup_distribution = dist = klass(attrs)
  File "C:\Users\scrut\code\env35\lib\site-packages\setuptools\dist.py", line 272, in __init__
    _Distribution.__init__(self,attrs)
  File "c:\users\scrut\appdata\local\programs\python\python35-32\Lib\distutils\dist.py", line 281, in __init__
    self.finalize_options()
  File "C:\Users\scrut\code\env35\lib\site-packages\setuptools\dist.py", line 327, in finalize_options
    ep.load()(self, ep.name, value)
  File "c:\users\scrut\code\phi\auth\.eggs\setuptools_scm-1.11.0-py3.5.egg\setuptools_scm\integration.py", line 19, in version_keyword
  File "c:\users\scrut\code\phi\auth\.eggs\setuptools_scm-1.11.0-py3.5.egg\setuptools_scm\__init__.py", line 104, in get_version
  File "c:\users\scrut\code\phi\auth\.eggs\setuptools_scm-1.11.0-py3.5.egg\setuptools_scm\__init__.py", line 82, in _do_parse
LookupError: setuptools-scm was unable to detect version for 'C:\\Users\\scrut\\code\\phi\\auth'.

Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.

For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
```

**Comment by RonnyPfannschmidt** [comment]
that gives no further idea, im under the impression something doesnt happen on windows

does the repo have any tags at all? (if not, then setuptools_scm should report something like `0.1.devXXX`)

**Comment by mr-rodgers** [comment]
RIght so I pulled the tags from GitHub and I'm still getting this:

```
(env35) C:\Users\scrut\code\phi\auth [develop +0 ~4 -0 | +0 ~16 -0 !]> git describe
v0.2.3
(env35) C:\Users\scrut\code\phi\auth [develop +0 ~4 -0 | +0 ~16 -0 !]> python setup.py egg_info
root 'C:\\Users\\scrut\\code\\phi\\auth'
looking for ep setuptools_scm.parse_scm C:\Users\scrut\code\phi\auth
found ep .git = setuptools_scm.git:parse
cmd 'git rev-parse --show-toplevel'
out b'C:/Users/scrut/Code/phi/auth\n'
real root C:/Users/scrut/Code/phi/auth
Traceback (most recent call last):
  File "setup.py", line 58, in <module>
    entry_points={},
  File "c:\users\scrut\appdata\local\programs\python\python35-32\Lib\distutils\core.py", line 108, in setup
    _setup_distribution = dist = klass(attrs)
  File "C:\Users\scrut\code\env35\lib\site-packages\setuptools\dist.py", line 272, in __init__
    _Distribution.__init__(self,attrs)
  File "c:\users\scrut\appdata\local\programs\python\python35-32\Lib\distutils\dist.py", line 281, in __init__
    self.finalize_options()
  File "C:\Users\scrut\code\env35\lib\site-packages\setuptools\dist.py", line 327, in finalize_options
    ep.load()(self, ep.name, value)
  File "c:\users\scrut\code\phi\auth\.eggs\setuptools_scm-1.11.0-py3.5.egg\setuptools_scm\integration.py", line 19, in version_keyword
  File "c:\users\scrut\code\phi\auth\.eggs\setuptools_scm-1.11.0-py3.5.egg\setuptools_scm\__init__.py", line 104, in get_version
  File "c:\users\scrut\code\phi\auth\.eggs\setuptools_scm-1.11.0-py3.5.egg\setuptools_scm\__init__.py", line 82, in _do_parse
LookupError: setuptools-scm was unable to detect version for 'C:\\Users\\scrut\\code\\phi\\auth'.

Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.
```

I'll test this on my Linux setup later with this same repo and report whether it works there, just to confirm that this is indeed a Windows thing.

**Comment by RonnyPfannschmidt** [comment]
unfortunately i currently dont have a windows bug, i'm not sure how to test/verify yet

**Comment by mr-rodgers** [comment]
Hi, after some digging around, I found what this issue was: https://github.com/pypa/setuptools_scm/blob/f5390e98b426da854838738c86a48b87742e259d/setuptools_scm/git.py#L15

real_root and root are different on my system, because `C:\Users\scrut\code` on the filesystem is actually `C:\Users\scrut\Code`. `git rev-parse --show-toplevel` capitalizes `Code` to match the canonical name, while `abspath` called earlier doesn't (I'm assuming because the console directory is lowercased -- I'm lazy). On Windows, this distinction shouldn't actually matter as the filesystem is case-insensitive. The check I linked to then gives an incorrect result in this case, as both paths differ in case for that segment, but point to the same directory.

I've patched it on my end using `os.path.normcase` before the check. I can submit a pull request if you wish.
